### ISSUE TITLE
Heuristic/loose completions for the 'any' type

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#5334]
# Motivation

Today, if you have a value of type 'any', dotting off of the value and requesting members for completion will return nothing back. The original motivation for this was that we should never risk giving users incorrect completions.

This is great for a fully typed TypeScript project. However, for those migrating to TypeScript, as well as those who need to drop down to `any` fairly frequently, this can be slightly frustrating.

For those used to the JS editing experience in editors like Sublime where they just "got completions", the current behavior might be seen as unattractive.
# Proposal

We should consider adding a language service option for users to get "loose" completions akin to what we give in the Salsa language service. 

We can still keep the warning for each completion, but we should provide completions with a builder.

**Comment by Zorgatone** [comment]
Yeah, it should be useful. Like other normal JavaScript IDE are doing (ie. WebStorm and similars...)

**Comment by mhegazy** [comment]
We have something similar in JS file completions, i wounder if that is what we should enable if we run in an `any`.

**Comment by DanielRosenwasser** [comment]
That's precisely what I mean. :smiley:

**Comment by RyanCavanaugh** [comment]
This happens now

**Comment by DanielRosenwasser** [comment]
Not in a TypeScript file it doesn't.

**Comment by RyanCavanaugh** [comment]
![image](https://user-images.githubusercontent.com/6685088/62102671-9ea02780-b24e-11e9-8c91-01749d47df41.png)

**Comment by basarat** [comment]
https://github.com/microsoft/TypeScript/issues/5334#issuecomment-516265598 suspect this isn't TypeScript powered but vscode text-word-reuse (not actual name) powered

**Comment by DanielRosenwasser** [comment]
Yup, VS Code will give word completions in a `.ts` file, but only if you explicitly trigger it. You can see the difference if you use a comment.

![image](https://user-images.githubusercontent.com/972891/62109538-576f6200-b261-11e9-93d1-764c646cc319.png)

![image](https://user-images.githubusercontent.com/972891/62109568-66eeab00-b261-11e9-9408-c6816099c485.png)

That said...VS Code's word completions are pretty good. I requested this back when Visual Studio was our primary editor. Maybe we should roll with the current behavior?
